pkg/agent/controller/networkpolicy/fqdn.go

```diff
@@ -594,38 +637,54 @@ func (f *fqdnController) runRuleSyncTracker(stopCh <-chan struct{}) {
 }
 
 // parseDNSResponse returns the FQDN, IP query result and lowest applicable TTL of a DNS response.
-func (f *fqdnController) parseDNSResponse(msg *dns.Msg) (string, map[string]net.IP, uint32, error) {
+func (f *fqdnController) parseDNSResponse(msg *dns.Msg) (string, map[string]ipWithTTL, error) {
 	if len(msg.Question) == 0 {
-		return "", nil, 0, fmt.Errorf("invalid DNS message")
+		return "", nil, fmt.Errorf("invalid DNS message")
 	}
 	fqdn := strings.ToLower(msg.Question[0].Name)
-	lowestTTL := uint32(math.MaxUint32) // a TTL must exist in the RRs
-	responseIPs := map[string]net.IP{}
+	// Case 1: In the upcoming patch an admin would set this to a  maximum value, which will be read from the configuration.
```

Comment:
note that in with the minTTL feature, expirationTime will be `maxTime(minTTL, r.Header().Ttl` instead. However, without a minTTL set, it should still be `minTime(maxConfiguredTTL, r.Header().Ttl)`

Reply:
@Dyanngg  Get it now, and indeed i was actually getting a little confused as i was trying to put the upcoming patch related modifications here, which i understand now, will happen in upcoming days when i start working on the patch itself.